tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15672,6 +15672,49 @@ void TryAddShaderAttrFromTargetProfile(Sema &S, FunctionDecl *FD,
   return;
 }
 
+// The compiler should emit a warning when an entry-point-only attribute
+// is detected without the presence of a shader attribute,
+// to prevent reliance on deprecated behavior
+// (where the compiler would infer a specific shader kind based on
+// a present entry-point-only attribute).
+void WarnOnEntryAttrWithoutShaderAttr(Sema &S, FunctionDecl *FD) {
+  if (FD->hasAttrs()) {
+    for (Attr *A : FD->getAttrs()) {
+      switch (A->getKind()) {
+        // Entry-Function-only attributes
+      case clang::attr::HLSLClipPlanes:
+      case clang::attr::HLSLDomain:
+      case clang::attr::HLSLEarlyDepthStencil:
+      case clang::attr::HLSLInstance:
+      case clang::attr::HLSLMaxTessFactor:
+      case clang::attr::HLSLNumThreads:
+      case clang::attr::HLSLRootSignature:
+      case clang::attr::HLSLOutputControlPoints:
+      case clang::attr::HLSLOutputTopology:
+      case clang::attr::HLSLPartitioning:
+      case clang::attr::HLSLPatchConstantFunc:
+      case clang::attr::HLSLExperimental:
```

Comment:
[experimental] attribute is used to develop experimental features.  It is not an entry point-specific attribute.